islandora.module

```diff
@@ -1685,8 +1690,12 @@ function islandora_object_access($op, $object, $user = NULL) {
  */
 function islandora_islandora_object_access($op, $object, $user) {
   module_load_include('inc', 'islandora', 'includes/utilities');
-
-  return islandora_namespace_accessible($object->id) && user_access($op, $user);
+  if ($object->state != 'A') {
+    return islandora_namespace_accessible($object->id) && user_access($op, $user) && user_access(ISLANDORA_ACCESS_INACTIVE_AND_DELETED_OBJECTS, $user);
```

Comment:
@fuel37 @rosiel @jordandukart @dannylamb would it be feasible to address



>  If an administrator at institution A wants to create a new role they need to be aware of this new permission and grant it. Similarly, if there are permissions bundled together in features, or another configuration control, they would need to be updated to grant this new permission to these roles.

 to do the following:

a simple user_role_grant_permissions() inside a https://api.drupal.org/api/drupal/modules%21user%21user.api.php/function/hook_user_role_insert/7.x to make sure even new roles get the "expected and current" behavior if we add a something like a  "use legacy expected behavior something for inactive-deleted" switch? Forget my wording, but you get the idea.